daemon/info_unix.go

```diff
@@ -57,29 +58,64 @@ func (daemon *Daemon) fillPlatformInfo(v *system.Info, sysInfo *sysinfo.SysInfo,
 	v.ContainerdCommit.ID = "N/A"
 	v.InitCommit.ID = "N/A"
 
-	if _, _, commit, err := parseDefaultRuntimeVersion(&cfg.Runtimes); err != nil {
-		log.G(context.TODO()).Warnf(err.Error())
-	} else {
+	if err := func() error {
+		_, _, commit, err := parseDefaultRuntimeVersion(&cfg.Runtimes)
+		if err != nil {
+			log.G(ctx).WithError(err).Warn("Failed to retrieve default runtime version")
+			return nil
+		}
 		v.RuncCommit.ID = commit
+		return nil
+	}(); err != nil {
+		return err
 	}
 
-	if rv, err := daemon.containerd.Version(context.Background()); err == nil {
+	if err := func() error {
```

Comment:
Wondering if it would be cleaner to define actual functions for these instead of all the closures. Perhaps this function is doing too much, and splitting up in smaller functions would make the code more readable, and easier to maintain.